### PR TITLE
Theme refactor

### PR DESCRIPTION
**Pull request by IntelOrca**
I have changed the theme format to JSON I think that will stand better than INI. This now removes a lot of the INI IO duplication and replaces a lot of the theme management with C++ code. Some of it is still hacky whilst the option and theme window remain in C.

This has allowed me to extend the core OpenRCT2 utility library further.

### CODE REVIEW

File: src/core/Json.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+{
+    json_t * ReadFromFile(const utf8 * path, size_t maxSize)
+    {
+        json_t  * json = nullptr;
+        auto fs = FileStream(path, FILE_MODE_OPEN);
+
+        size_t fileLength = (size_t)fs.GetLength();
+        if (fileLength > maxSize)
+        {
+            throw IOException("Language file too large.");
```
Comment (IntelOrca) [inline src/core/Json.cpp]: String should say JSON.

File: src/audio/mixer.cpp [hunk 2]
```diff
@@ -667,6 +668,10 @@ void Mixer::MixChannel(Channel& channel, uint8* data, int length)
 				volumeadjust *= (gConfigSound.master_volume / 100.0f);
 				if (channel.group == MIXER_GROUP_MUSIC) {
 					volumeadjust *= (gConfigSound.music_volume / 100.0f);
+				} else {
+					if (RCT2_GLOBAL(RCT2_ADDRESS_SCREEN_FLAGS, uint8) & SCREEN_FLAGS_TITLE_DEMO) {
+						volumeadjust *= 0.5f;
+					}
```
Comment (IntelOrca) [inline src/audio/mixer.cpp]: Take this out, included unintentionally in this PR.

**Comment by janisozaur** [comment]
So we have `jansson` and custom one now? Why?

**Comment by IntelOrca** [comment]
What do you mean, custom one?

**Comment by janisozaur** [comment]
Oh, nevermind. I just skimmed through the code and noticed new `Json.cpp`, assumed it was using own JSON parsing. Now I see I was wrong, sorry.

File: src/core/List.hpp [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+    void Add(T item)
+    {
+        this->push_back(item);
+    }
+
+    void Insert(T item, size_t index)
+    {
+        Guard::ArgumentInRange(index, (size_t)0, this->size());
+        this->insert()
```
Comment (zsilencer) [inline src/core/List.hpp]: seems like this wouldn't actually insert anything

**Comment by IntelOrca** [comment]
@janisozaur could you help me fix the GCC errors?
@LRFLEW could you modify the xcode project?

**Comment by danidoedel** [comment]
Maybe an extra string id could be used for "Player" without a colon to be consistent with the other strings?
theme

**Comment by danidoedel** [comment]
If you duplicate a theme, the original theme gets the same name. If you then try to delete the original theme the game crashes.

theme

**Comment by danidoedel** [comment]
Looks good :+1:

**Pull request merged**